### PR TITLE
CloudQuery-GCP C2C doc

### PR DESCRIPTION
**Pull request by kimsauce**
## Purpose of this pull request

This pull request adds CloudQuery GCP C2C doc and CID.

## Select the type of change

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)

https://sumologic.atlassian.net/browse/CONN-3157

### CODE REVIEW

**Comment by kimsauce** [comment]
Hi @JV0812 , can I hand this off to you? Left some notes in the Jira ticket and your Google Doc.

**Comment by JV0812** [comment]
Sure @kimsauce, I will take up this PR. Thanks for working on this.

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+id: cloudquery-gcp-source
+title: CloudQuery GCP (Beta)
+sidebar_label: CloudQuery GCP
+tags:
+  - cloud-to-cloud
+  - cloudquery
+  - gcp
+  - google cloud platform
+description: Learn how to collect inventory from the GCP APIs using CloudQuery Plugin SDK and send it to Sumo Logic.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:10]:
```suggestion
description: Learn how to collect inventory from GCP APIs using the CloudQuery Plugin SDK and send it to Sumo Logic.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+import CodeBlock from '@theme/CodeBlock';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import ExampleJSON from '/files/c2c/cloudquery-gcp/example.json';
+import MyComponentSource from '!!raw-loader!/files/c2c/cloudquery-gcp/example.json';
+import TerraformExample from '!!raw-loader!/files/c2c/cloudquery-gcp/example.tf';
+
+<img src={useBaseUrl('img/send-data/cloudquery-logo.png')} alt="cloudquery-icon" width="50" />
+
+The CloudQuery GCP integration is to pull inventory from various Google Cloud Platform (GCP) APIs via the CloudQuery GCP plugin, transform them into the CloudQuery schema, and then send it to Sumo Logic.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:27]:
```suggestion
The CloudQuery GCP integration pulls inventory from various Google Cloud Platform (GCP) APIs via the CloudQuery GCP plugin, transforms it into the CloudQuery schema, and then sends it to Sumo Logic.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+## Setup
+
+### Prerequisites
+
+[Enable](https://cloud.google.com/endpoints/docs/openapi/enable-api) all the service APIs in the authenticated GCP account related with the services from which you want to fetch data via CloudQuery GCP plugin.
+
+### Vendor configuration
+
+Download the user credentials JSON file by using the gcloud CLI, refer to the [Google Cloud Documentation](https://cloud.google.com/docs/authentication/application-default-credentials#personal).
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:47]:
```suggestion
To download the user credentials JSON file by using the Google Cloud CLI, refer to the [Google Cloud documentation](https://cloud.google.com/docs/authentication/application-default-credentials#personal).
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+### Source configuration
+
+When you create an CloudQuery GCP source, you add it to a Hosted Collector. Before creating the source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector](/docs/send-data/hosted-collectors/configure-hosted-collector).
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:51]:
```suggestion
When you create a CloudQuery GCP source, you add it to a Hosted Collector. Before creating the source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector and Source](/docs/send-data/hosted-collectors/configure-hosted-collector).
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+1. **(Optional) Project ID(s)**. Enter the unique identifier number for your project. You can find this from the Google Cloud Console. If not specified, all the available active projects will be considered by default.
+1. **(Optional) Organization ID(s)**. Enter the unique identifier number for your organization. You can find this from the Google Cloud Console. If not specified, all the available active projects will be considered by default.
+1. **Services**. Enter the type of service from which the GCP data should be fetched via CloudQuery GCP plugin.
+1. By default, **Polling Interval** is set to 12 hours.
+1. **(Optional) Processing Rules for Logs**. Configure any desired filters, such as allowlist, denylist, hash, or mask, as described in [Create a Processing Rule](/docs/send-data/collection/processing-rules/create-processing-rule). 
+1. When you are finished configuring the Source, click **Save**.
+
+## JSON schema
+
+Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [Use JSON to configure Sources](/docs/send-data/use-json-configure-sources) for details.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:68]:
```suggestion
Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [Use JSON to Configure Sources](/docs/send-data/use-json-configure-sources) for details. 
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+## JSON schema
+
+Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [Use JSON to configure Sources](/docs/send-data/use-json-configure-sources) for details. 
+
+| Parameter | Type | Required | Description |
+|:---|:---|:---|:---|
+| config | JSON Object | Yes | Contains the configuration parameters for the source |
+| schemaRef | JSON Object | Yes | Use `{“type”: “CloudQuery GCP Inventory”}` |
+| sourceType | String | Yes | Use “Universal” for CloudQuery GCP source Inventory |
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:74]:
```suggestion
| config | JSON Object | Yes | Contains the configuration parameters for the source. |
| schemaRef | JSON Object | Yes | Use `{“type”: “CloudQuery GCP Inventory”}.` |
| sourceType | String | Yes | Use “Universal” for CloudQuery GCP source Inventory. |
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Message
+
+```
+transport: received unexpected content-type \"text/html; charset=UTF-8
+```
+
+#### Solution
+
+If you come across this warning message while fetching service data for the configured source then please contact CloudQuery or GCP team.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:112]:
```suggestion
If you come across this warning message while fetching service data for the configured source, then contact the CloudQuery or GCP team.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+### Error handling
+
+For the SDK, if the response code is `401` or `403`, this is config error. You'll need to quit the integration.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:116]:
```suggestion
For the SDK, if the response code is `401` or `403`, it indicates a config error. You'll need to quit the integration.
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+For any other error returned by CloudQuery, we will set ErrorGeneric as an error in C2C. Integration will not stop its execution.
```
Comment (jpipkin1) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:118]:
```suggestion
For any other error returned by CloudQuery, we will set ErrorGeneric as an error in the integration. The integration will not stop its execution.
```

**Review by jpipkin1 — APPROVED** [review]
@JV0812, I approved the PR contingent on your addressing my suggestions.

**Comment by JV0812** [comment]
@jpipkin1 Thanks for your review, I have incorporated your comments.

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+1. When you are finished configuring the Source, click **Save**.
+
+## JSON schema
+
+Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [Use JSON to Configure Sources](/docs/send-data/use-json-configure-sources) for details. 
+
+| Parameter | Type | Required | Description |
+|:---|:---|:---|:---|
+| config | JSON Object | Yes | Contains the configuration parameters for the source. |
+| schemaRef | JSON Object | Yes | Use `{“type”: “CloudQuery GCP Inventory”}.` |
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:73]:
```suggestion
| schemaRef | JSON Object | Yes | Use `{“type”: “CloudQuery GCP Inventory”}`. |
```

File: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+transport: received unexpected content-type \"text/html; charset=UTF-8
+```
+
+#### Solution
+
+If you come across this warning message while fetching service data for the configured source, then contact the CloudQuery or GCP team.
+
+### Error handling
+
+For the SDK, if the response code is `401` or `403`, it indicates a config error. You'll need to quit the integration.
```
Comment (kimsauce) [inline docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/cloudquery-gcp-source.md:116]:
```suggestion
For the SDK, if the response code is `401` or `403`, it indicates a config error. You'll need to quit and restart the integration.
```

**Comment by jpipkin1** [comment]
@kimsauce , we’re removing the CloudQuery GCP Source beta article with PR [#4209](https://github.com/SumoLogic/sumologic-documentation/pull/4209) at the request of @Arpit Jaiswal (see [Slack](https://sumologic.slack.com/archives/C063N5N2C5U/p1723674224220399)).

**Pull request merged**